### PR TITLE
AW - Auth Challenge

### PR DESCRIPTION
**Pull request by wagnerAJS221**
I havent got the last movie reqs working as of yet.

### CODE REVIEW

File: src/client/App.jsx [hunk 1]
```diff
@@ -1,56 +1,122 @@
...
+        console.log('this the data console', data)
+      })
+  }
 
   const handleLogin = async ({ username, password }) => {
-    
-  };
-  
+    const body = {
+      username: username,
```
Comment (lucianosimoni) [inline src/client/App.jsx:47]: just the `username` would be nice, same as line 22 ☕

File: src/client/App.jsx [hunk 2]
```diff
@@ -1,56 +1,122 @@
...
+    }
+
+    fetch('http://localhost:4000/user/login', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(body)
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        const keyName = Object.keys(data)[0]
```
Comment (lucianosimoni) [inline src/client/App.jsx:58]: I've also used KeyName conditions 😁

File: src/client/App.jsx [hunk 3]
```diff
@@ -1,56 +1,122 @@
...
+          localStorage.setItem('LoginToken', data.token)
+        } else {
+          setLoginResponse('Error: Username or Password are invalid!')
+        }
+      })
+  }
+
   const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
+    const elToken = localStorage.getItem(token)
```
Comment (lucianosimoni) [inline src/client/App.jsx:72]: nice cabrón 🤠 HAHSHAH

File: src/server/controllers/movie.js [hunk 4]
```diff
@@ -1,31 +1,44 @@
-const jwt = require('jsonwebtoken');
-const { PrismaClient } = require('@prisma/client');
-const prisma = new PrismaClient();
+const jwt = require('jsonwebtoken')
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
 
-const jwtSecret = 'mysecret';
+const jwtSecret = 'mysecret'
```
Comment (lucianosimoni) [inline src/server/controllers/movie.js:5]: you could use the **.env** to store these values ☕ Would be nicer in a real-world project

File: src/client/App.jsx (hunk 3 above)
Comment (wagnerAJS221) [reply src/client/App.jsx:72]: Muchas Gracias senor!

File: src/server/controllers/user.js [hunk 5]
```diff
@@ -1,39 +1,62 @@
...
+
+  bcrypt.hash(password, saltrounds, async (err, hashedPwd) => {
+    try {
+      const createdUser = await prisma.user.create({
+        data: {
+          username: username,
+          password: hashedPwd
+        }
+      })
+      delete createdUser.password
```
Comment (lucianosimoni) [inline src/server/controllers/user.js:21]: I've learned that this `delete` keyword belongs to the Object constructor in JavaScript.
Nice to see it being used here to delete a key of `createdUser` obj

**Pull request closed without merging**